src/rc_string.c

```diff
@@ -65,29 +66,41 @@ static void rc_string_dispose(RC_STRING* content)
 
 static THANDLE(RC_STRING) rc_string_create_impl(const char* string)
 {
+    THANDLE(RC_STRING) result = NULL;
+
     /* Codes_SRS_RC_STRING_01_002: [ Otherwise, rc_string_create shall determine the length of string. ]*/
     size_t string_length = strlen(string);
-    size_t string_length_with_terminator = string_length + 1;
-
-    /* Codes_SRS_RC_STRING_01_003: [ rc_string_create shall allocate memory for the THANDLE(RC_STRING), ensuring all the bytes in string can be copied (including the zero terminator). ]*/
-    THANDLE(RC_STRING) temp_result = THANDLE_MALLOC_WITH_EXTRA_SIZE(RC_STRING)(rc_string_dispose, sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + string_length_with_terminator);
-    if (temp_result == NULL)
+    size_t memory_allocation_max_size = SIZE_MAX - (sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + 1);
+    if (string_length > memory_allocation_max_size)
```

Comment:
>memory_allocation_max_size [](http://example.com/codeflow?start=24&length=26)

This could be replaced here with SIZE_MAX - (sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + 1

No need for an extra variable.